Source/VMware.vSphereDSC/Tests/Integration/VMHostService.Integration.Tests.ps1

```diff
@@ -0,0 +1,241 @@
+)
+
+$script:dscResourceName = 'VMHostService'
+$script:dscConfig = $null
+$script:moduleFolderPath = (Get-Module VMware.vSphereDSC -ListAvailable).ModuleBase
+$script:integrationTestsFolderPath = Join-Path (Join-Path $moduleFolderPath 'Tests') 'Integration'
+$script:configurationFile = "$script:integrationTestsFolderPath\Configurations\$($script:dscResourceName)\$($script:dscResourceName)_Config.ps1"
+
+$script:configWithServicePolicyUnset = "$($script:dscResourceName)_WithServicePolicyUnset_Config"
+$script:configWithServicePolicyOn = "$($script:dscResourceName)_WithServicePolicyOn_Config"
+$script:configWithServicePolicyOff = "$($script:dscResourceName)_WithServicePolicyOff_Config"
+$script:configWithServicePolicyOnAndRunning = "$($script:dscResourceName)_WithServicePolicyOnAndRunning_Config"
+
+$script:connection = Connect-VIServer -Server $Server -User $User -Password $Password
+$script:vmHost = $null
+
+$script:Key = 'TMS-SSH'
+$script:PolicyUnset = 'Unset'
+$script:PolicyOff = 'Off'
+$script:PolicyOn = 'On'
+$script:RunningFalse = $false
+$script:RunningTrue = $true
+
+$script:resourceWithServicePolicyUnset = @{
+    Name = $Name
+    Server = $Server
+    Key = $script:Key
+    Policy = $script:PolicyUnset
+    Running = $script:RunningFalse
+}
+$script:resourceWithServicePolicyOn = @{
+    Name = $Name
+    Server = $Server
+    Key = $script:Key
+    Policy = $script:PolicyOn
+    Running = $script:RunningFalse
+}
+$script:resourceWithServicePolicyOff = @{
+    Name = $Name
+    Server = $Server
+    Key = $script:Key
+    Policy = $script:PolicyOff
+    Running = $script:RunningFalse
+}
+$script:resourceWithServicePolicyOnAndRunning = @{
+    Name = $Name
+    Server = $Server
+    Key = $script:Key
+    Policy = $script:PolicyOn
+    Running = $script:RunningTrue
+}
+
+. $script:configurationFile -Name $Name -Server $Server -User $User -Password $Password
+
+$script:mofFileWithServicePolicyUnset = "$script:integrationTestsFolderPath\$($script:configWithServicePolicyUnset)\"
+$script:mofFileWithServicePolicyOn = "$script:integrationTestsFolderPath\$($script:configWithServicePolicyOn)\"
+$script:mofFileWithServicePolicyOff = "$script:integrationTestsFolderPath\$($script:configWithServicePolicyOff)\"
+$script:mofFileWithServicePolicyOnAndRunning = "$script:integrationTestsFolderPath\$($script:configWithServicePolicyOnAndRunning)\"
+
+Describe "$($script:dscResourceName)_Integration" {
+    Context "When using configuration $($script:configWithServicePolicyUnset)" {
+
+        It 'Should compile and apply the MOF without throwing' {
+            # Assert
+            {
+                $startDscConfigurationParameters = @{
+                    Path = $script:mofFileWithServicePolicyUnset
+                    ComputerName = 'localhost'
+                    Wait = $true
+                    Force = $true
+                }
+                Start-DscConfiguration @startDscConfigurationParameters
+            } | Should -Not -Throw
```

Comment:
Here we should have the following: Start-DscConfiguration @startDscConfigurationParameters | Should -Not -Throw ( And for all other Start-DscConfiguration Tests the same applies).

Reply:
Not sure I understand this remark.
What is the problem with
```powershell
{
   $startDscConfigurationParameters = @{
      Path = $script:mofFileWithServicePolicyUnset
     ComputerName = 'localhost'
     Wait = $true
     Force = $true
  }
  Start-DscConfiguration @startDscConfigurationParameters
} | Should -Not -Throw
```
Doesn't an eventual throw from Start-DscConfiguration propagate from the code block?

Reply:
Yes. this is still valid. Because all other Integration Tests were written using:
```
 Start-DscConfiguration @startDscConfigurationParameters | Should -Not -Throw
```
And I think for consistency is good to write the tests this way.
Another reason is that this way shows explicitly which expression potentially throws.